addons/website_forum/models/forum_post.py

```diff
@@ -789,6 +790,90 @@ def _notify_thread_by_inbox(self, message, recipients_data, msg_vals=False, **kw
     # WEBSITE
     # ----------------------------------------------------------------------
 
+    def get_microdata(self):
+        """
+        Generate structured data (microdata) for the post.
+
+        Returns:
+            str or False: Microdata in JSON format representing the post, or False if not applicable.
+        """
+        # Return False if it's not a question.
+        if self.parent_id:
+            return False
+        # Retrieve the possible correct answer to this question
+        post = self.child_ids.filtered(lambda x: x.is_correct)
+        correct_post = {
+            "text": post[0].plain_content,
```

Comment:
I prevent that `with correct_post = { ... } if post else False`

Reply:
I see, I overlooked it, but can't we do this?
Instead of
```py
        correct_post = {
            "text": post[0].plain_content,
            "upvoteCount": post[0].vote_count,
            "url": post[0].website_url,
            "datePublished": post[0].create_date.strftime("%Y-%m-%dT%H:%M:%SZ"),
            "author_name": post[0].create_uid.name,
            "author_url": f"/forum/user/{ post[0].create_uid.id }",
        } if post else False
        
        if correct_post:
            structured_data["mainEntity"]["acceptedAnswer"] = {
                "@type": "Answer",
                "text": correct_post["text"],
                "upvoteCount": correct_post["upvoteCount"],
                "datePublished": correct_post["datePublished"],
                "url": correct_post["url"],
                "author": {
                    "@type": "Person",
                    "name": correct_post["author_name"],
                    "url": correct_post["author_url"],
                }
            }
```
Just do
```py
        if post:
            structured_data["mainEntity"]["acceptedAnswer"] = {
                "@type": "Answer",
                "text": post[0].plain_content,
                "upvoteCount": post[0].vote_count,
                "datePublished": post[0].create_date.strftime("%Y-%m-%dT%H:%M:%SZ",
                "url": post[0].website_url,
                "author": {
                    "@type": "Person",
                    "name": post[0].create_uid.name,
                    "url": f"/forum/user/{ post[0].create_uid.id }",
                }
            }
```

The `correct_post` actually holds no purpose. And on top of that, it's constructing a dict of values to then "deconstruct" it into another dict